packages/anatomy/src/anatomy.ts

```diff
@@ -0,0 +1,32 @@
+export const anatomy = (component: string) => {
+  let _component = component
+  let _parts: string[]
+
+  const api = {
+    parts<T extends string>(...parts: T[]) {
+      if (_parts) {
+        throw new Error('anatomy("").parts(...) should only be called once. Did you mean to use .extend(...) ?')
+      }
+      _parts = parts
+      return api
+    },
+
+    extend<T extends string>(...parts: T[]) {
+      _parts = [..._parts, ...parts]
+      return api
+    },
+
+    build() {
+      return [...new Set(_parts)].reduce<Record<string, { selector: string }>>(
+        (prev, part) =>
+          Object.assign(prev, {
+            [part]: {
+              selector: `[data-scope="${_component}"][data-part="${part}"]`,
```

Comment:
I would love to add these attributes as well to reduce the need for manually adding these to every component.

```suggestion
              selector: `[data-scope="${_component}"][data-part="${part}"]`,
              attrs: { "data-scope":  _component, "data-part": part }
```